sources/web/datalab/authentication.ts

```diff
@@ -57,37 +57,29 @@ function getAccessTokenFromMetadataServer(cb: common.Callback<string>) {
   }
 }
 
-function lookupUserFromDatastore(userId: string, accessToken: string, cb: common.Callback<boolean>) {
-  var postData: any = {
-    keys: [
-      {
-        path: [
-          {
-            kind: 'DatalabUser',
-            name: userId
-          }
-        ]
-      }
-    ]
-  };
-  var dataStorePath: string = '/datastore/v1beta2/datasets/' + appSettings.projectId + '/lookup';
+function lookupUserFromStorage(userId: string, accessToken: string, cb: common.Callback<boolean>) {
+  var bucketId = appSettings.projectId + '-datalab';
+  var storagePath: string = '/storage/v1/b/' + bucketId + '/o/users%2F' + userId;
   try {
-    httpapi.posts('www.googleapis.com', dataStorePath, null, postData, accessToken, null, function(e: Error, data: any) {
+    httpapi.gets('www.googleapis.com', storagePath, null, accessToken, null, function(e: Error, data: any) {
       if (e) {
-        logging.getLogger().error(e, 'Failed to query Datastore for authentication.');
-        cb && cb(e, null);
+        if (e.message.length >= 3 && e.message.substr(e.message.length - 3) == '404') {
+          // Most of times we cannot tell whether it is a transient error or the object not found,
+          // because they both return 404. But we assume it is caused by the object not exist.
+          cb && cb(null, false);
```

Comment:
This can be simplified just a bit, as I think cb is always going to be non-null, right?